### PR TITLE
Exclusive XInput event handling

### PR DESCRIPTION
**Pull request by Caellian**
This commit builds upon #1819 and makes conky use _only_ XInput for mouse events if `BUILD_MOUSE_EVENTS` is enabled.

This PR makes conky only propagate basic X11 mouse events, because constructing XInput events is a bit tricky (or impossible):
- We can't just forward the event we've received because XInput stores event details in a cookie which can't be claimed multiple times.
- We can't pass them through and ignore them.
- I'm still looking into ways of constructing XInput events.

### CODE REVIEW

**Comment by brndnmtthws** [comment]
Do you have an ETA on when you might be finished this? I was thinking of cutting a new release in the next few days, but I will wait if you will be done soon.

**Comment by Caellian** [comment]
> Do you have an ETA on when you might be finished this?

I expect within the following 2-3 days. There's a quirk I need to figure out a workaround for (working on that currently), other than that it's pretty much done. The PR doesn't introduce any functionality though, it just switches to a newer event system on X11.

File: src/x11.cc [hunk 1]
```diff
@@ -307,6 +310,57 @@ __attribute__((noreturn)) static int x11_ioerror_handler(Display *d) {
...
+// Polyfill overwrite default Xlib definitions to account for virtual root
+// windows
+#undef RootWindowOfScreen
+#define RootWindowOfScreen(s) VirtualRootWindowOfScreen(s)
+#undef RootWindow
+#define RootWindow(dpy, screen) \
+  VirtualRootWindowOfScreen(ScreenOfDisplay(dpy, screen))
+#undef DefaultRootWindow
+#define DefaultRootWindow(dpy) \
+  VirtualRootWindowOfScreen(DefaultScreenOfDisplay(dpy))
```
Comment (Caellian) [inline src/x11.cc:362]: Moved handling of virtual roots from `find_desktop_window` here and overwrote Xlib macros to use it for consistency.

Overwriting macros means that reading `wm-spec` atoms requires accessing (real) root though `screen->root`, so I'm not 100% sure about doing that.

File: src/x11.cc [hunk 2]
```diff
@@ -574,54 +574,49 @@ void destroy_window() {
   memset(&window, 0, sizeof(struct conky_x11_window));
 }
 
-void x11_init_window(lua::state &l, bool own) {
```
[the remark this answers is not in the document]
Comment (Caellian) [reply src/x11.cc:577]: Moving these changes to a separate PR.

**Review by brndnmtthws — APPROVED** [review]

**Comment by jborme** [comment]
This commit causes me trouble. With fluxbox as WM, a right click on the root window leads to 3  delay waiting for the WM right-click menu to appear, with 100% CPU in the meantime.

**Comment by Caellian** [comment]
You can build with `BUILD_XINPUT` disabled for now if that's the case. Thanks for reporting this, I will be able to investigate and fix it in a week or so.

Time to install yet another X11 WM... :rocket:

**Pull request merged**